lib/galaxy/model/store/__init__.py

```diff
@@ -2213,8 +2213,19 @@ def add_dataset_collection(
         self.included_collections.append(collection)
 
     def add_dataset(self, dataset: model.DatasetInstance, include_files: bool = True) -> None:
+        if include_files:
+            self._ensure_dataset_file_accessible(dataset)
         self.included_datasets[dataset] = (dataset, include_files)
 
+    def _ensure_dataset_file_accessible(self, dataset: model.DatasetInstance) -> None:
+        is_sessionless = dataset.id is None
+        state = dataset.dataset.state
+        # Sessionless datasets are not persisted in the object store, so we can skip the check as there is no file_path associated with them.
```

Comment:
That doesn't seem right as a general statement, pushing datasets to the object store is the export models responsibility, but that happens after calling `add_dataset`

Reply:
Thank you as always Marius! That is what I needed to know.
I was quite confused as I expected it to be already persisted at this point. I will move the check to the proper place :+1: